packages/file-search/src/browser/quick-file-open.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from "inversify";
+import { QuickOpenModel, QuickOpenItem, QuickOpenMode, QuickOpenService, OpenerService } from '@theia/core/lib/browser';
+import { FileSystem, FileStat } from '@theia/filesystem/lib/common/filesystem';
+import { FileIconProvider } from '@theia/filesystem/lib/browser/icons/file-icons';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+import URI from '@theia/core/lib/common/uri';
+import { FileSearchService } from '../common/file-search-service';
+import { CancellationTokenSource } from '@theia/core/lib/common';
+
+@injectable()
+export class QuickFileOpenService implements QuickOpenModel {
+
+    constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
+        @inject(FileIconProvider) protected readonly fileIconProvider: FileIconProvider,
+        @inject(OpenerService) protected readonly openerService: OpenerService,
+        @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
+        @inject(FileSearchService) protected readonly fileSearchService: FileSearchService
+    ) {
+        workspaceService.root.then(root => this.wsRoot = root);
```

Comment:
-1 I'm still not convinced about the 'user-home' use case.

Reply:
Thinking more about this I think this could wait for the workspace indeed as it is different than a quick Open File, it's more a quick search file.

So OK. 

I sure would like to have a Quick Open File/Directory however using the command palette I'll open an issue about that.